carbon-design-system/carbon-addons-iot-react
# V3 upgrade

Closes #

## This is being merged into V3

**Summary**
It looks like a lot--and it is--but the majority of it is the eslinting upgrade changes and changing `bx--` and `iot--` prefixes to use constants. The items that specifically had merge conflicts are listed here and should be the ones checked more thoroughly. There were merge conflicts with:
- src/index.js - linting change for `export {default ` versus items that are not being exported in v3. TableHead, TableBody, EmptyTable, TableWithSkeletonHeader
- src/components/SuiteHeader.jsx - href as `#` instead of `javascript:void(0)`
- src/components/Table/Table.jsx - adding shouldExpandOnRowClick prop removal of wrapText/truncateText props
- src/components/Table/Table.test.jsx - removing tests related to wrapText/truncateText
- src/components/Table/TablePropTypes.js - adding new props to end of file
- src/components/Table/TableBody.jsx - adding preserveCellWhiteSpace prop
- - src/components/Table/TableBodyRow.jsx - adding preserveCellWhiteSpace prop
- src/components/Table/TableCellRenderer.test.jsx - adding preserveCellWhiteSpace prop / removing wrapText/truncateText tests
- src/components/Table/TableHead/TableHead.jsx - adding hasRowNesting prop, columnGrouping code

**Acceptance Test (how to verify the PR)**

- check public api snaphots for inadvertent removals.
- check table stories (specifically related to these features: preservingWhiteSpace, columnGrouping, and cell truncation)

**Regression Test (how to make sure this PR doesn't break old functionality)**

- see above



**Things to look for during review**
